src/network/networkd-link.c

```diff
@@ -2638,11 +2681,23 @@ int link_update(Link *link, sd_netlink_message *m) {
                 r = link_add(manager, m, &link);
                 if (r < 0)
                         return r;
+
+                hashmap_remove_value(link->manager->links_by_name, link->ifname, link);
+                hashmap_put(link->manager->links_by_name, ifname, link);
```

Comment:
oom not handled. maybe better move this into link_drop() and link_add() and make the latter fail if this doesn't work?

Reply:
OK